test/config-luatest/vars_test.lua

```diff
@@ -0,0 +1,307 @@
+-- * {{ replicaset_name }}
+-- * {{ group_name }}
+g.test_basic = function(g)
+    local cases = {
+        {
+            var_name = 'instance_name',
+            var_value = 'instance-001',
+        },
+        {
+            var_name = 'replicaset_name',
+            var_value = 'replicaset-001',
+        },
+        {
+            var_name = 'group_name',
+            var_value = 'group-001',
+        },
+    }
+
+    -- Choose some string option to set it to {{ var_name }}
+    -- and read the value from inside the instance (in verify()
+    -- function).
+    local option = 'process.title'
+
+    for _, case in ipairs(cases) do
+        helpers.success_case(g, {
+            options = {
+                [option] = ('{{ %s }}'):format(case.var_name),
+            },
+            verify = function(option, var_value)
+                local config = require('config')
+                t.assert_equals(config:get(option), var_value)
+            end,
+            verify_args = {option, case.var_value},
+        })
+    end
+end
+
+-- Verify several template variables within one option.
+g.test_several_vars = function(g)
+    local option = 'process.title'
+
+    helpers.success_case(g, {
+        options = {
+            [option] = ('{{ %s }}::{{ %s }}::{{ %s }}'):format('group_name',
+                'replicaset_name', 'instance_name')
+        },
+        verify = function(option)
+            local config = require('config')
+            t.assert_equals(config:get(option), ('%s::%s::%s'):format(
+                'group-001', 'replicaset-001', 'instance-001'))
+        end,
+        verify_args = {option},
+    })
+end
+
+-- Verify that template variables are correctly calculated for
+-- replicaset peers.
+--
+-- The configdata module depends on box.cfg(), so we can't run
+-- this unit test directly. Let's run it as a script.
+g.test_peer = run_as_script(function()
+    local cluster_config = require('internal.config.cluster_config')
+    local configdata_lib = require('internal.config.configdata')
+    local t = require('myluatest')
+
+    local listen_template = table.concat({'unix/:.', 'var', 'run',
+        '{{ group_name }}', '{{ replicaset_name }}', '{{ instance_name }}',
+        'tarantool.iproto'}, '/')
+
+    local instance_name = 'instance-001'
+    local cconfig = {
+        iproto = {
+            listen = listen_template,
+        },
+        groups = {
+            ['group-001'] = {
+                replicasets = {
+                    ['replicaset-001'] = {
+                        instances = {
+                            ['instance-001'] = {},
+                            ['instance-002'] = {},
+                        },
+                    },
+                },
+            },
+        },
+    }
+    local iconfig = cluster_config:instantiate(cconfig, instance_name)
+    local configdata = configdata_lib.new(iconfig, cconfig, instance_name)
+
+    local function exp_uri(group_name, replicaset_name, instance_name)
```

Comment:
This function is a one-time use, so maybe it's easier to use a local variable: `local exp = table.concat(...)`?

Reply:
My idea is to make the test cases constructed in a similar way (so it is easier to spot actual differences).